pkg/config/parse.go

```diff
@@ -0,0 +1,99 @@
+package config
+
+import (
+	"io/ioutil"
+	"runtime"
+
+	"github.com/BurntSushi/toml"
+	"github.com/kelseyhightower/envconfig"
+	validator "gopkg.in/go-playground/validator.v9"
+)
+
+var validate *validator.Validate
+
+func getValidator() *validator.Validate {
+	if validate == nil {
+		return validator.New()
+	}
+	return validate
+}
+
+// Config provides configuration values for all components
+type Config struct {
+	GoEnv                string         `validate:"required" envconfig:"GO_ENV" default:"development"`
+	GoBinary             string         `validate:"required" envconfig:"GO_BINARY_PATH" default:"go"`
+	LogLevel             string         `validate:"required" split_words:"true" default:"debug"`
+	MaxConcurrency       int            `validate:"required" split_words:"true"`
+	MaxWorkerFails       uint           `validate:"required" split_words:"true" default:"5"`
+	CloudRuntime         string         `validate:"required" split_words:"true" default:"none"`
+	FilterFile           string         `validate:"required" split_words:"true" default:"filter.conf"`
+	Timeout              int            `validate:"required" default:"300"`
+	EnableCSRFProtection bool           `envconfig:"ATHENS_ENABLE_CSRF_PROTECTION" default:"false"`
+	Proxy                *ProxyConfig   `validate:""`
+	Olympus              *OlympusConfig `validate:""`
+	Storage              *StorageConfig `validate:""`
+}
+
+// ParseConfigFile parses the given file into an athens config struct
+func ParseConfigFile(configFile string) (*Config, error) {
+
+	// attempt to read the given config file
```

Comment:
`envconfig` supports reading from environment or `yml` file. Any reason to have a `toml` file?

Reply:
I didn't think [envconfig](https://github.com/kelseyhightower/envconfig) has support for yaml files. Am I missing something?

No specific reason for toml over yaml, I just personally prefer it for configs.

The idea with this change was to move towards using config files over only environment variables so we could have a few distinct configs (dev/test/prod) that would be easy to switch between and reason about

Reply:
Oops, my bad. yaml support is in viper. If its plain file with `export`, running `go test` in different directory would be simpler. toml works too.